### DISCUSSION TITLE
Where to install SonarQube plugin JAR file?

### DISCUSSION THREAD

**Discussion by shuting-zhang-gh** [Homebrew/discussions#3435] [Everyday usage]
My system: Mac OS Big Sur 11.2.3

I wanted to add a plugin to SonarQube that is not on its marketplace.

According to SonarQube's doc here: https://docs.sonarqube.org/latest/setup/install-plugin/, I should **"Put the downloaded jar in $SONARQUBE_HOME/extensions/plugins".**

**But there is no `/extensions/plugins` folders** after installing SonarQube with Homebrew. 

My SonarQube folder only has three subfolders:
<img width="379" alt="image" src="https://user-images.githubusercontent.com/39432060/179379391-cb64fd47-05d8-4f2c-8013-2f7197f32e71.png">

I tried 3 ways to install this plugin but it didn't work. 

1. Create `/extensions/plugins/` folders and put the JAR file there. It didn't work because the plugin didn't appear on SonarQube's server web page.

2. Put the JAR file in `/usr/local/var/sonarqube/data/web/deploy/plugins/bitegarden-sonarqube-metrics-plugin-1.2.jar`, because I discovered that installed-via-marketplace-plugins are here. But it didn't work either. After restarting the SonarQube server, the folder and the contained JAR file were gone.

3. Created a folder in the above path and put the JAR file there: `/usr/local/var/sonarqube/data/web/deploy/plugins/bitegarden/bitegarden-sonarqube-metrics-plugin-1.2.jar`, since I saw that all plugins were contained in a folder. It still didn't work. After restarting the SonarQube server, the folder and the contained JAR file were gone.

I saw that in the path `usr/local/var/sonarqube/data/web/deploy/plugins/`, there is a file `index.txt` which contains information of each installed plugin, there are their name, a boolean value, JAR file's name, then a long alphanumeric string which looks like an id. 
<img width="399" alt="image" src="https://user-images.githubusercontent.com/39432060/179379516-1432a821-e24a-4282-8aae-0d09fd1b0eff.png">

I would add a line myself in `index.txt` of the plugin I want to install, but I don't know what to add for the last field that looks like an id.

Any help is appreciated!

**Comment by SMillerDev** [comment]
That sounds like a question to the sonarqube developers since I have no idea what `$SONARQUBE_HOME` resolves to

**Comment by Kelly-Fisher-27Global** [comment]
Did you ever find a solution to this? I'd like to install a few plugins, but can't find out how to add them to a HomeBrew installed instance

**Comment by AwonSomeSauce** [comment]
I have installed SonarQube via Homebrew and its extensions are stored in `/opt/homebrew/Cellar/sonarqube/10.0.0.68432/libexec/extensions/plugins`. I believe that copying the .jar file of the desired extension in this directory will suffice. I hope this helps!
